android/app/src/main/java/au/com/robin/sms/MainActivity.kt

```diff
@@ -0,0 +1,69 @@
+package au.com.robin.sms
+
+import android.Manifest
+import android.content.Intent
+import android.content.pm.PackageManager
+import android.os.Bundle
+import android.util.Log
+import android.widget.Toast
+import androidx.appcompat.app.AppCompatActivity
+import androidx.core.app.ActivityCompat
+import androidx.core.content.ContextCompat
+import au.com.robin.sms.databinding.ActivityMainBinding
+
+class MainActivity : AppCompatActivity() {
+    private lateinit var smsInterfaceBinding: ActivityMainBinding
+    private val PERMISSION_REQUEST_CODE = 1
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        smsInterfaceBinding = ActivityMainBinding.inflate(layoutInflater)
+        val view = smsInterfaceBinding.root
+        setContentView(view)
+
+        smsInterfaceBinding.btSend.setOnClickListener {
+            val phoneNo = smsInterfaceBinding.etPhoneNo.text?.toString()
```

Comment:
In `activity_main.xml` the id is `et_phoneNo` with the underscore. They need to match right?

Reply:
I'm not a fan of Android's XML UI stuff. So don't do too much of it.

Reply:
> In `activity_main.xml` the id is `et_phoneNo` with the underscore. They need to match right?

This is because the use of viewBinding. It converts it to camel case.  
<br>

> I'm not a fan of Android's XML UI stuff. So don't do too much of it. Just the bare minimum.

Do you mean to just use findViewById and just dump all the views just like initialising variables?
e.g.
```kotlin
// at the top
var textView: TextView
var editText: EditText
```
something like that

Reply:
Ok I don't like the fact that `et_phoneNo` does not match. I meant just not to care too much about UI code. But what you mention about `findViewById` or creating TextView/EditView programmatically, sounds more appealing to me.

Reply:
In my past Android project,  I did not create any xml files for UI. But not high priority.

Reply:
I see, so you want it to be created programatically. I will keep that in mind